ArmoniK.SDK.Client/src/ChannelPool.cpp

```diff
@@ -2,20 +2,86 @@
 
 #include <armonik/common/options/ControlPlane.h>
 #include <grpcpp/create_channel.h>
+#include <shared_mutex>
 #include <utility>
 
 namespace SDK_CLIENT_NAMESPACE::Internal {
 
-std::shared_ptr<grpc::Channel> ChannelPool::GetChannel() {
-  // TODO Setup TLS / mTLS support
-  // TODO Setup channel recycling
+std::shared_ptr<grpc::Channel> ChannelPool::AcquireChannel() {
+  std::shared_ptr<grpc::Channel> channel;
+  if (channel_pool_.size() != 0) {
+    channel = channel_pool_.front();
+    channel_pool_.pop();
```

Comment:
Lock should be moved up here, as it's only useful for the channels container, not the rest

Reply:
You can use a temporary scope for that